.github/workflows/rust-ci-long-test.yml

```diff
@@ -0,0 +1,126 @@
+name: Rust CI - Long Test (Reusable)
+
+on:
```

Comment:
@jorgemmsilva Is this just a template it will it auto run?

Reply:
it's just so it's reusable (to avoid copy/pasting the same contents to the migration e2e test workflow). 
Initially we bundled them together, but it makes a little more sense to have them separated and run only when necessary